web/test/import.test.ts

```diff
@@ -8,35 +8,109 @@
 
-  beforeEach(() => {})
-  it('should do something', () => {
-    migrationData = JSON.stringify(sampleGoodDataExport)
-    internalImportProjectsData(
+  let whoami: any
+  let mockBaseRootState: any
+  let mockGetState: any
+
+  let mockMigrationData: any
+
+  // if you want a quick way to get at the type expressed in the signature
+  // of a function you can use something similar to: ReturnType<typeof ProfilesApi>
+  // aah nice, whats the ReturnType<> for?
+  // it means if this is a function, what type does it return, e.g. Promise<SomeType>
+  // gotcha. nice!
+
+  beforeEach(() => {
+    getAppWs = jest.fn().mockResolvedValue({})
+    createProfilesZomeApi = jest.fn().mockReturnValue({
+      profile: {
+        // turn this jest.fn into a var that you can track and assert on
+        createWhoami: jest.fn(),
+      },
+    })
+    installProjectAppAndImport = jest.fn()
+    installProjectApp = jest
+      .fn()
+      .mockResolvedValue(['testCellIdString', ['abc'], 'testString'])
+    onStep = jest.fn()
+    whoami = {
+      actionHash: 'testActionHash',
+      entryHash: 'testEntryHash',
+      createdAt: 1234, // nanoseconds
+      updatedAt: 1234, // nanoseconds
+      entry: {
+        firstName: 'testFirstName',
+        lastName: 'testLastName',
+        handle: 'testHandle',
+        status: 'Online',
+        avatarUrl: 'testAvatarUrl',
+        agentPubKey: 'testAgentPubKey',
+        isImported: false,
+      },
+    }
+    mockBaseRootState = {
+      whoami,
+      cells: {
+        profiles:
+          '132,45,36,204,129,221,8,19,206,244,229,30,210,95,157,234,241,47,13,85,105,207,55,138,160,87,204,162,244,122,186,195,125,254,5,185,165,224,66[:cell_id_divider:]132,32,36,97,138,27,24,136,8,80,164,189,194,243,82,224,72,205,215,225,2,27,126,146,190,40,102,187,244,75,191,172,155,196,247,226,220,92,1',
+        projects: [
+          '132,45,36,67,75,209,140,160,204,62,71,45,229,66,99,63,6,255,250,52,234,238,45,50,174,198,118,29,208,28,207,156,147,252,58,99,131,165,51[:cell_id_divider:]132,32,36,97,138,27,24,136,8,80,164,189,194,243,82,224,72,205,215,225,2,27,126,146,190,40,102,187,244,75,191,172,155,196,247,226,220,92,1',
+        ],
+      },
+    }
+    mockMigrationData = JSON.stringify(sampleGoodDataExport)
+
+    mockGetState = jest.fn().mockReturnValue(mockBaseRootState)
+
+    store = {
+      dispatch: jest.fn(),
+      getState: mockGetState,
+    }
+  })
+
+  it('should do something', async () => {
+    await internalImportProjectsData(
       getAppWs,
       createProfilesZomeApi,
       installProjectAppAndImport,
+      installProjectApp,
       store,
-      migrationData,
+      mockMigrationData,
       onStep
     )
+
+    expect(getAppWs).toHaveBeenCalledTimes(1)
+    expect(createProfilesZomeApi).toHaveBeenCalledTimes(1)
+    expect(installProjectAppAndImport).toHaveBeenCalledTimes(1)
+    expect(installProjectApp).toHaveBeenCalledTimes(1)
+    expect(store.dispatch).toHaveBeenCalled()
```

Comment:
I'd like to see these three get assertions like 'toHaveBeenCalledWith' and to assert what the args to those calls were

Reply:
`toHaveBeenCalledWith` is just rough. I don't know what the actual method is, but I am looking for the one that lets you assert the args to the call